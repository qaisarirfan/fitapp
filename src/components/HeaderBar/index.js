import React from "react"
import {View, Text} from "react-native"
import PropTypes from "prop-types"
import {useNavigation} from "@react-navigation/native"
import Entypo from "react-native-vector-icons/Entypo"
import {TouchableOpacity} from "react-native-gesture-handler"
import themeStyles from "./styles"
import {CustomStatusBar} from "../CustomStatusBar"

// HeaderBar Component content
export const HeaderBar = ({title, isHeaderShow, rightContent}) => {
  const styles = themeStyles()
  const {canGoBack, goBack} = useNavigation()

  let style = {}
  if (isHeaderShow) {
    style = {
      // backgroundColor: styleableTheme[800],
    }
  } else {
    style = {
      paddingBottom: 0,
      paddingTop: 0,
    }
  }

  return (
    <View style={[styles.root, {...style}]}>
      <CustomStatusBar
        barStyle="light-content"
        // backgroundColor={styleableTheme[800]}
      />
      <View style={[styles.header, {...style}]}>
        <View style={styles.back}>
          {isHeaderShow && canGoBack() ? (
            <TouchableOpacity
              testID="back"
              onPress={goBack}
              style={styles.backButton}>
              <Entypo
                name="chevron-left"
                size={16}
                // color={styleableTheme[50]}
              />
              <Text style={styles.text}>Back</Text>
            </TouchableOpacity>
          ) : null}
        </View>
        {isHeaderShow ? (
          <View style={styles.title}>
            <Text ellipsizeMode="tail" numberOfLines={1} style={styles.text}>
              {title}
            </Text>
          </View>
        ) : null}
        <View style={styles.right}>{rightContent}</View>
      </View>
    </View>
  )
}

// HeaderBar Proptypes
HeaderBar.propTypes = {
  isHeaderShow: PropTypes.bool,
  title: PropTypes.string.isRequired,
  rightContent: PropTypes.node,
}

// HeaderBar Default props
HeaderBar.defaultProps = {
  isHeaderShow: true,
  rightContent: null,
}

export default HeaderBar
