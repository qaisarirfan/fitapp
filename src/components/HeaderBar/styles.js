import {Dimensions, StyleSheet} from "react-native"

// HeaderBar: Styling here
const styles = () => {
  const screen = Dimensions.get("screen")
  const width = screen.width / 3
  return StyleSheet.create({
    root: {},
    header: {
      display: "flex",
      flexGrow: 1,
      flexDirection: "row",
      justifyContent: "space-between",
      paddingTop: 12,
      paddingBottom: 12,
    },
    back: {
      width,
      justifyContent: "center",
      alignItems: "flex-start",
      paddingLeft: 6,
    },
    backButton: {
      display: "flex",
      flexDirection: "row",
    },
    title: {
      width,
      alignItems: "center",
      justifyContent: "center",
      // backgroundColor: theme[800],
    },
    text: {
      // color: theme[50],
      fontSize: 18,
      lineHeight: 18,
    },
    right: {
      width,
      alignItems: "flex-end",
      justifyContent: "center",
      paddingRight: 6,
    },
  })
}

export default styles
