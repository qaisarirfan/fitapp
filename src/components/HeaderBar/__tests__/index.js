import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import HeaderBar from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<HeaderBar />", () => {
  const createProps = () => ({
    title: "test",
  })
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<HeaderBar {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
