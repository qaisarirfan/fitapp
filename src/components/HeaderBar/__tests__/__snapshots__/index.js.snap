// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar /> Does not explode 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "#36505d",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 20,
        },
        Object {
          "backgroundColor": "#36505d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexGrow": 1,
          "justifyContent": "space-between",
          "paddingBottom": 12,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#36505d",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "justifyContent": "center",
          "paddingLeft": 6,
          "width": 250,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="back"
      >
        <View
          accessible={true}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#eaedef",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï…‚
          </Text>
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 18,
                "lineHeight": 18,
              }
            }
          >
            Back
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#36505d",
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#eaedef",
            "fontSize": 18,
            "lineHeight": 18,
          }
        }
      >
        test
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "paddingRight": 6,
          "width": 250,
        }
      }
    />
  </View>
</View>
`;
